xilem/src/property_tuple.rs

```diff
@@ -154,7 +82,45 @@ macro_rules! __declare_property_tuple_loop {
             $Type: ident, $idx: tt;
         )+
     ) => {
-        type $Props = ($(Option<$Type>,)+);
+
+        #[expect(missing_docs)]
+        pub struct $Props($(Option<$Type>,)+);
```

Comment:
Why is this `pub`? I don't think this type is going to be part of the public API of any widget, nor have any rules.
I think it should at least have some basic docs if you're going to make it so.
Or it should be `expect(unreachable_pub)`, if it's for the associated type on `Style`

Reply:
I think this comment still stands - I'd like to understand why this type is pub.

Reply:
Oh, right, forgot to answer. Sorry.

It needs to be pub because it's used as an associated type for the various implementations of the `Style` trait, and you're not allowed to do that with a private type. We could do a pub-in-private trick or something similar, but I think just making it a pub item works for now.